app/models/metric/ci_mixin/capture.rb

```diff
@@ -80,10 +80,12 @@ def perf_capture_queue(interval_name, options = {})
       :state       => ['ready', 'dequeue'],
     }
 
+    messages = MiqQueue.where.not(:method_name => 'perf_capture_realtime').where(queue_item).index_by(&:args)
```

Comment:
We want to add more fields here to use an index.

There are 2 possible indexes, but I'm thinking the second one is better:

- queue_name, zone, role, server_guid, priority, deliver_on, task_id (WHERE state = 'ready')
- class_name, method_name, queue_name, state, zone

Please change `where.not(:method_name => '')` => `where(:method_name => %w())`